numpy/lib/function_base.py

```diff
@@ -990,15 +997,36 @@ def gradient(f, *varargs, **kwargs):
     >>> y = x**2
     >>> np.gradient(y, dx, edge_order=2)
     array([-0.,  2.,  4.,  6.,  8.])
+
+    The axis keyword can be used to specify a subset of axes of which the gradient is calculated
+    >>> np.gradient(np.array([[1, 2, 6], [3, 4, 5]], dtype=np.float), axis=0)
+    array([[ 2.,  2., -1.],
+           [ 2.,  2., -1.]])
     """
     f = np.asanyarray(f)
     N = len(f.shape)  # number of dimensions
+
+    axes = kwargs.pop('axis', tuple(range(N)))
+    # check axes to have correct type and no duplicate entries
+    if isinstance(axes, int):
+        axes = (axes,)
+    if not isinstance(axes, tuple):
+        raise TypeError("A tuple of integers or a single integer is required")
```

Comment:
Unless I am missing something, this code will error on an explicit `None` passed in by the user, which is documented as being equivalent as "over all axes."

My suggestion would be to have the default value when you `.pop` `'axis'` be `None`, then immediately check if `axes` is `None` and have it replaced by `tuple(range(N))`: it kind of makes the code tell the story both of `None` being the default, and of it meaning all axes.